microsoft/testsuites/cvm/cvm_boot.py

```diff
@@ -0,0 +1,176 @@
+from typing import Any, List, cast
+
+from assertpy.assertpy import assert_that, assert_warn
+
+from lisa import (
+    Environment,
+    Logger,
+    Node,
+    RemoteNode,
+    TestCaseMetadata,
+    TestSuite,
+    TestSuiteMetadata,
+)
+from lisa.features.security_profile import (
+    CvmEnabled,
+    SecurityProfile,
+    SecurityProfileSettings,
+)
+from lisa.operating_system import CBLMariner, Posix
+from lisa.sut_orchestrator import AZURE
+from lisa.testsuite import TestResult, simple_requirement
+from lisa.tools import Lsblk, Reboot, Tpm2
+from lisa.tools.lsblk import PartitionInfo
+from lisa.util import (
+    SkippedException,
+    TcpConnectionException,
+    UnsupportedDistroException,
+    constants,
+)
+from lisa.util.shell import wait_tcp_port_ready
+
+
+@TestSuiteMetadata(
+    area="cvm",
+    category="functional",
+    description="""This test suite covers some common scenarios related to
+    CVM boot on Azure.
+    """,
+)
+class CVMBootTestSuite(TestSuite):
+    def before_case(self, log: Logger, **kwargs: Any) -> None:
+        node: Node = kwargs["node"]
+        if not isinstance(node.os, CBLMariner):
+            raise SkippedException(
+                UnsupportedDistroException(
+                    node.os, "CVM boot test supports only Azure Linux."
+                )
+            )
+
+    @TestCaseMetadata(
+        description="""This test verifies that TPM enrollment is done correctly on
+        a CVM with encrypted root partition
+        """,
+        priority=2,
+        requirement=simple_requirement(
+            supported_features=[CvmEnabled()],
+            supported_platform_type=[AZURE],
+        ),
+    )
+    def verify_encrypted_root_partition(
+        self,
+        log: Logger,
+        node: RemoteNode,
+        environment: Environment,
+        log_path: Path,
+        result: TestResult,
+    ) -> None:
+        security_profile_settings = cast(
+            SecurityProfileSettings, node.features[SecurityProfile].get_settings()
+        )
+        if not security_profile_settings.encrypt_disk:
+            raise SkippedException("This test requires disk encryption to be enabled")
+        lsblk = node.tools[Lsblk]
+        disks = lsblk.get_disks(force_run=True)
+        partitions: List[PartitionInfo] = next(
+            (d.partitions for d in disks if d.name == "sda"), []
+        )
+        assert_that(partitions, "Cannot find a disk named 'sda'").is_not_empty()
+        root_partition = next((p for p in partitions if p.name == "sda2"), None)
+        assert_that(root_partition, "Cannot locate root partition").is_not_none()
+        assert isinstance(root_partition, PartitionInfo)
+        assert_that(root_partition.fstype).is_equal_to("crypto_LUKS")
+
+    @TestCaseMetadata(
+        description="""This test case verifies that a CVM can still boot if any boot
+        component is upgraded.
+
+        Steps:
+        1. On first boot, check current PCR values for PCR4 and PCR7
```

Comment:
Add requirements on new environment, if it's needed.

Reply:
I have defined requirement for this test case below (CvmEnabled, AZURE). Does this suffice?

Reply:
Because the steps writes "on **first** boot". If some test coverage needs a fresh new environment, it needs a flag to make sure the test case runs first on a new deployed environment.

Reply:
I see. It looks like a new feature to add to Lisa. I was only thinking of controlling it inside a runbook via `keep_environment = false`. Do you have any suggestions on how I should approach it?

Reply:
I think it should be this one 

https://github.com/microsoft/lisa/blob/main/microsoft/testsuites/core/azure_image_standard.py#L887